src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Order/TreeItemOrderPropertyProviderSource.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Order
+{
+    /// <summary>
+    /// Setup the dataflow for a provider that updates solution tree item properties with 
+    /// display order metadata derived from IOrderedSourceItemsDataSourceService
+    /// </summary>
+    [Export(typeof(IProjectTreePropertiesProviderDataSource))]
+    [AppliesTo(ProjectCapability.SortByDisplayOrder)]
+    internal class TreeItemOrderPropertyProviderSource : ChainedProjectValueDataSourceBase<IProjectTreePropertiesProvider>, IProjectTreePropertiesProviderDataSource
+    {
+        [Import(ExportContractNames.Scopes.UnconfiguredProject)]
+        private IOrderedSourceItemsDataSourceService OrderedItemSource { get; set; }
+
+        [ImportingConstructor]
+        public TreeItemOrderPropertyProviderSource(UnconfiguredProject project)
+            : base(project.Services)
+        {
+        }
+
+        protected override IDisposable LinkExternalInput(ITargetBlock<IProjectVersionedValue<IProjectTreePropertiesProvider>> targetBlock)
+        {
+            JoinUpstreamDataSources(OrderedItemSource);
+
+            TreeItemOrderPropertyProvider latestTreeItemOrderPropertyProvider = null;
```

Comment:
>latestTreeItemOrderPropertyProvider [](start = 42, length = 35)

I would move this variable to be a class field

Reply:
What is the advantage of making this a class field?

---
In reply to: [148048938](https://github.com/dotnet/project-system/pull/2930#discussion_r148048938) [](ancestors = 148048938)

Reply:
For clarity and to have a way to make it thread safe?

---
In reply to: [148100953](https://github.com/dotnet/project-system/pull/2930#discussion_r148100953) [](ancestors = 148100953,148048938)